### PR TITLE
Update C# style guide

### PR DESCRIPTION
**Pull request by MythicManiac**
* Add instructions for no region usage
 * Some developers prefer using regions and some don't, but organizing the code in a way that it's clean without regions works for both parties.
 * Not using regions encourages bettere code practices, such as splitting larger functions into smaller ones or huge classes into multiple smaller classes/files.
* Add instructions for dictionary/switch/if-else statements
 * Instead of using long if/else statements or switches, Dictionaries are often a better choice
 * Dictionaries are usually the most optimal way of implementing logic in terms of execution speed, extensibility, modification, code length and readability.
 * **TL;DR**  If you feel like a switch should be used, Dictionary is probably a better option.
* Add instructions for boolean variable naming
 * Consistently naming booleans in the same way will make reading the code easier
* Add instructions for conditional operator (`condition ? result1 : result2`)
 * Should be avoided as much as possible
 * Alright in some niche cases where it's required and remains readable

### CODE REVIEW

**Comment by MythicManiac** [comment]
Feel free to give a thumbs up/down, although if you give a thumbs down, leave a comment on why.

**Comment by molenzwiebel** [comment]
I'd argue that using the ternary operator for simple variable switches `team == RED ? redSpawns : blueSpawns` is fine. Embedding more logic into them should be avoided though.
